packages/blocks/src/api/parser/convert-legacy-block.js

```diff
@@ -77,5 +77,33 @@ export function convertLegacyBlockNameAndAttributes( name, attributes ) {
 		newAttributes.legacy = true;
 	}
 
+	// Convert pattern overrides added during experimental phase.
```

Comment:
This logic is only necessary in the Gutenberg plugin so it can be put behind the check:

https://github.com/WordPress/gutenberg/blob/82e22bf55e01b56d330130eb3fe2fcb48ee8c681/tools/webpack/shared.js#L67

This way it won't get included in WordPress core.